pkg/apiserver/apiserver.go

```diff
@@ -370,7 +371,24 @@ func write(statusCode int, apiVersion string, codec runtime.Codec, object runtim
 		io.Copy(writer, out)
 		return
 	}
-	writeJSON(statusCode, codec, object, w, isPrettyPrint(req))
+	exporter, exact := getExporter(req)
+	exportJSON(statusCode, codec, object, w, isPrettyPrint(req), exporter, exact)
+}
+
+func getExporter(req *http.Request) (export.Exporter, bool) {
+	exp := req.URL.Query().Get("export")
```

Comment:
If we did leave this as a query param, we'd want to handle it similarly to other query params. See DeleteOptions and ListOptions as examples.